src/core/server.cc

```diff
@@ -806,6 +806,7 @@ InferenceServer::HandleInfer(
     OnCompleteInferRPC();
   };
 
+  backend->Backend()->SetInferenceServer(this);
```

Comment:
You should put a comment here: "Need to set 'this' in each backend even though it is redundant after the first time. Once we remove TFS dependency we can construct each backend in a way that makes it directly aware of the inference server"